### DISCUSSION TITLE
Incepting Http-Request

### DISCUSSION THREAD

**Discussion by iamgabrielsoft** [LawsonCheng/poke#110] [Ideas]
The package 📦 should be able to intercept http request from users. Before forwarding it to 'POST', 'GET', 'PUT' methods

**Comment by LawsonCheng** [comment]
Hello, any example usage suggested, and may I know why we need to intercept the http request?

**Comment by iamgabrielsoft** [comment]
Well I thought about having a header method(Inceptor) where the request or response is received first before forwarding it to the user...


For now no example exempt for the Axios package Inceptor method

**Reply by LawsonCheng** [reply]
Do you mean the returned result including a property say "headers"?

**Comment by iamgabrielsoft** [comment]
Yes a property headers for either request or response sent to the server...

**Comment by iamgabrielsoft** [comment]
The feature is quite similar to axios inceptor, if you could spare some
time to have a look at it on axios documentation that would be great...

**Comment by iamgabrielsoft** [comment]
I would be implementing it
